### PR TITLE
python3Packages.rasterio: 1.1.5 → 1.2.6

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
[Changelog](https://github.com/mapbox/rasterio/blob/master/CHANGES.txt)

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123589` at debff1d0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/d73ec1125e0f52275801c2ed715e9143">python38Packages.rasterio</a></li>
    <li><a href="https://gist.github.com/970b06fbc58878e017083b0b4f2cf6c2">python39Packages.rasterio</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `packaging`.
Near pkgs/development/python-modules/rasterio/default.nix:4:22:
```
  |
4 | , pytest, pytestcov, packaging, hypothesis, boto3
  |                      ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by jonringer — COMMENTED** [review]
please address test errors:
```
       last 10 log lines:
       >     from shapely.geometry import shape
       > ModuleNotFoundError: No module named 'shapely'
       >
       >
       > ----------------------------------------------------------------------
       > Ran 31 tests in 0.832s
       >
       > FAILED (errors=4)
       > Test failed: <unittest.runner.TextTestResult run=31 errors=4 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=31 errors=4 failures=0>
       For full logs, run 'nix log /nix/store/c3s9wf1q5nbws8li7p4k4702j4x935s0-python3.8-rasterio-1.2.3.drv'.
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python38Packages.rasterio

File: pkgs/development/python-modules/rasterio/default.nix [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
   src = fetchFromGitHub {
     owner = "mapbox";
     repo = "rasterio";
     rev = version;
-    sha256 = "168b6hmx026jsvhnq6s5k0qfhzda02mmx1alax6wqk16mk63mqcz";
+    sha256 = "sha256-0tq/0fba+/G9eWq2U1ERqCmEArTLDCiGn30DIJ8oC6g=";
   };
 
-  checkInputs = [ boto3 pytest pytestcov packaging hypothesis ] ++ lib.optional (!isPy3k) mock;
+  checkInputs = [ pytest pytestcov hypothesis shapely ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rasterio/default.nix:20]: this should be pytestCheckHook.

Comment (sikmir) [reply pkgs/development/python-modules/rasterio/default.nix:20]: It doesn't work for me with pytestCheckHook (`invalid module name: 'rasterio.errors'`).

Comment (SuperSandro2000) [reply pkgs/development/python-modules/rasterio/default.nix:20]: No idea. Maybe ask on the python nix matrix channel.

**Comment by sikmir** [comment]
Updated in #128806

**Pull request closed without merging**